src/promise.lua

```diff
@@ -170,6 +172,10 @@ function promise:new()
     p._state = PENDING
     p._value = nil
     p._reason = nil
+    p._rejectHandled = nil
+    local proxy = newproxy(true)
```

Comment:
Unfortunately, `newproxy` is [removed](https://stackoverflow.com/a/60329180/378298) in Lua 5.2+ :(

It's [easy enough](https://stackoverflow.com/a/60329532/378298) to make an alternative apparently, although not everything would be needed, just the `__gc` behavior things.

Reply:
with lua >= 5.2, __gc function is not anymore limited to userdata, we can add it to lua table. I will update the PR.